src/extension/report.ts

```diff
@@ -0,0 +1,50 @@
+import * as Sentry from "@sentry/node";
+import { v4 as uuidv4 } from "uuid";
+import {
+    workspace,
+} from "vscode";
+
+export class CrashReporter {
+    private enableCrashReporting: any;
+
+    constructor(
+        enableCrashReporting: boolean,
+    ) {
+        this.enableCrashReporting = enableCrashReporting;
```

Comment:
should we also figure out if crash reporting is on at this point?
then we can store that on the instance of the reporter instead
```typescript
        const telemetry = workspace.getConfiguration("telemetry");
        this.enableCrashReporting = telemetry.get("enableCrashReporter");
```